### DISCUSSION TITLE
Proper naming convention / casing convention

### DISCUSSION THREAD

**Discussion by torian257x** [livewire/livewire#3958] [1. Help] [answered]
it isn't transparent to me how the naming works

we have a namespace with all upper case e.g.

`app\LTS\etc`

gives me components like

`l-t-s.etc`

as well underscores seem to not work in properties

`$product_id` wont work

must be? `$productId`

is there somewhere a document that outlines all of these conventions?

**Comment by joshhanley** [comment, marked as answer]
@torian257x what version of Livewire are you using? As we added support for properties with an underscore in 2.6, so should work now.

Regarding class name versus view name, when you generate a component it converts the class namespace and name to kebab case for the views, as per Laravel's blade components convention, which converts CamelCased class names to kebab-cased view names.

Hope this helps!

**Reply by torian257x** [reply]
ah, I guess we are on 2.5, will check

And LTS is l-t-s because the rule is each capital letter is transformed to lower case and receives a dash with the special case of s not receiving one because its at the end of a namespace. Its not trivial... but I appreciate the answer a lot, thanks

**Reply by joshhanley** [reply]
Yeah exactly. No worries 🙂
